src/main.js

```diff
@@ -0,0 +1,105 @@
+import './style/main.scss'
+import React from 'react'
+import ReactDom from 'react-dom'
+import superagent from 'superagent'
+
+const API_URI = 'https://www.reddit.com/r'
+
+class RedditForm extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      data: '',
+    }
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleRedditQuery = this.handleRedditQuery.bind(this)
+  }
+
+  handleRedditQuery(e){
+    this.setState({data: e.target.value})
+  }
+
+  handleSubmit(e){
+    e.preventDefault()
+    this.props.redditSelect(this.state.data)
+  }
+
+  render(){
+    return (
+      <form onSubmit={this.handleSubmit} >
+        <input
+          type='text'
+          data='postTitle'
+          placeholder='Post Title'
+          value={this.state.data}
+          onChange={this.handleRedditQuery}
+          />
+      </form>
+    );
+  }
+}
+
+class App extends React.Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      redditTitle: null,
+      redditLookup: {},
+      redditSelected: null,
+      redditDataError: null,
+      }
+
+    this.redditSelect = this.redditSelect.bind(this)
+    }
+    componentDidUpdate(){
+      console.log('___STATE___', this.state);
+    }
+
+  redditSelect(data) {
+      console.log('about to get');
+      superagent.get(`${API_URI}/${data}.json`)
+      .then(res => {
+        console.log('selected post', res.body);
+        this.setState({
+          redditTitle: data,
+          redditSelected: res.body.kind,
+          redditDataError: null,
+        })
+      }).catch(() => {
+        this.setState({
+          redditSelected: null,
+          redditDataError: data,
+        })
+      });
+    }
+  render() {
+    return (
+      <div>
+        <h1>Reddit Request Form</h1>
+        <RedditForm redditSelect={this.redditSelect}/>
+      {this.state.redditDataError ?
+        <div>
+        <RedditForm/>
+          <h2>Subreddit {this.state.redditDataError} does not exist</h2>
+          <p>Try again !</p>
+        </div> :
+        <div>
+        {this.state.redditSelected ?
+          <div>
+          <RedditForm redditSelect={this.redditSelect}/>
+            <h2>Selected {this.state.redditTitle} Subreddit </h2>
```

Comment:
It looks like you are never returning the list of titles with their URL reference.
You want to map through each object so you can have access to the title and URL. I added some console logs so you can see what you are getting back at each stage.
```sh
<ul>
{this.state.redditSelected.map((item, i) => {
      console.log('ITEM::::', item)
      console.log('ITEM DATA TITLE::::', item.data.title)
      console.log('ITEM DATA URL::::', item.data.url);
      return (
          <li key={i}>
              <a href={item.data.url}> {item.data.title} </a>
           </li>
      )
  })}
</ul>